### DISCUSSION TITLE
[bug] route_class is ignored for websockets

### DISCUSSION THREAD

**Discussion by Tishka17** [fastapi/fastapi#11922] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.routing import APIRoute, APIRouter


class DishkaRoute(APIRoute):
    def __init__(
            self,
            path,
            *args,
            **kwargs,
    ):
        print("ROUTE:", path)
        super().__init__(path, *args, **kwargs)


router = APIRouter(route_class=DishkaRoute)


@router.websocket("/ws")
async def websocket_endpoint():
    pass


@router.get("/index")
async def index():
    pass
```

### Description

Expected output:
ROUTE: /ws
ROUTE: /index

Actual output:
ROUTE: /index

### Operating System

Linux

### FastAPI Version

0.111.1

### Pydantic Version

2.5.3

### Python Version

3.10.12

**Comment by Tishka17** [comment]
This bug affects [dishka](https://github.com/reagento/dishka/blob/develop/src/dishka/integrations/fastapi.py) autoinjection.

For get/post there is some logic in fastapi: https://github.com/fastapi/fastapi/blob/master/fastapi/routing.py#L883
While for websockets it relies fully on Starlette implementation - there is no implementation of `add_websocket_route` in fastapi

**Comment by Tishka17** [comment]
@tiangolo as it is required to have your confirmation to create an issue, I request your attention here. Should I create an issue?

**Comment by Tishka17** [comment]
Will be here any updates from maintainers?

**Comment by Kludex** [comment]
I think `route_class` will always be for non-websocket endpoints... Maybe a `websocket_class` can be added.
